polaris/ocean/tasks/isomip_plus/mesh/cull.py

```diff
@@ -0,0 +1,119 @@
+
+        Parameters
+        ----------
+        component : polaris.Component
+            The component this step belongs to
+
+        subdir : str
+            the subdirectory for the step
+
+        config : polaris.config.PolarisConfigParser
+            A shared config parser
+
+        base_mesh : polaris.Step
+            The base mesh step containing input files to this step
+
+        topo_remap : polaris.Step
+            The topography remapping step containing input files to this step
+        """  # noqa: E501
+        super().__init__(component=component, name='cull_mesh', subdir=subdir)
+        self.base_mesh = base_mesh
+        self.topo_remap = topo_remap
+        self.set_shared_config(config, link='isomip_plus.cfg')
+
+        base_path = base_mesh.path
+        target = os.path.join(base_path, 'base_mesh.nc')
+        self.add_input_file(filename='base_mesh.nc', work_dir_target=target)
+
+        topo_path = topo_remap.path
+        target = os.path.join(topo_path, 'topography_remapped.nc')
+        self.add_input_file(filename='topography.nc',
+                            work_dir_target=target)
+
+        for file in ['culled_mesh.nc', 'culled_graph.info']:
+            self.add_output_file(filename=file)
+
+    def run(self):
+        """
+        Run this step
+        """
+        logger = self.logger
+
+        # only use progress bars if we're not writing to a log file
+        use_progress_bar = self.log_filename is None
+
+        with LoggingContext(name=__name__, logger=logger) as logger:
+            _land_mask_from_topo(topo_filename='topography.nc',
+                                 mask_filename='land_mask.nc')
+
+            ds_base = xr.open_dataset('base_mesh.nc')
+            ds_land_mask = xr.open_dataset('land_mask.nc')
+
+            # cull the mesh based on the land mask
+            ds_culled = cull(ds_base, dsMask=ds_land_mask, logger=logger,
+                             dir='.')
+
+            # sort the cell, edge and vertex indices for better performances
+            ds_culled = sort_mesh(ds_culled)
+
+            add_isomip_plus_xy(ds_culled)
+
+            out_filename = 'culled_mesh.nc'
+            write_netcdf(ds_culled, out_filename)
+
+            # we need to make the graph file after sorting
+            make_graph_file(mesh_filename='culled_mesh.nc',
+                            graph_filename='culled_graph.info')
+
+            extract_vtk(ignore_time=True, dimension_list=['maxEdges='],
+                        variable_list=['allOnCells'],
+                        filename_pattern='culled_mesh.nc',
+                        out_dir='culled_mesh_vtk',
+                        use_progress_bar=use_progress_bar)
+
+
+def _land_mask_from_topo(topo_filename, mask_filename):
+    ds_topo = xr.open_dataset(topo_filename)
+
+    ocean_frac = ds_topo.oceanFraction
+
+    # we want the mask to be 1 where there's not ocean
+    mask = np.logical_or(ocean_frac < 0.5, ocean_frac.isnull())
```

Comment:
I don't think we need to make the min ocean fraction a config option, but can we make sure it's in the docs?

Reply:
It seems like the best way to document it is to make it a config option. I'll do that.

Reply:
Thanks!

Reply:
I added a config option.